app/lib/terra_analysis_service.rb

```diff
@@ -0,0 +1,52 @@
+# methods related to terra workflow analyses
+# corresponding to the 'analysis' tab in study overview
+class TerraAnalysisService
+
+  def self.list_submissions(study)
+    # load list of previous submissions
+    workspace = ApplicationController.firecloud_client.get_workspace(study.firecloud_project, study.firecloud_workspace)
+    submissions = ApplicationController.firecloud_client.get_workspace_submissions(study.firecloud_project, study.firecloud_workspace)
+
+    submissions.each do |submission|
+      update_analysis_submission(submission)
+    end
+    # remove deleted submissions from list of runs
+    if !workspace['workspace']['attributes']['deleted_submissions'].blank?
```

Comment:
Non-blocking: I think re-writing these to use `dig` would be more robust:
```suggestion
    if !workspace.dig('workspace', 'attributes','deleted_submissions').blank?
      deleted_submissions = workspace.dig('workspace', 'attributes', 'deleted_submissions', 'items')
```

Reply:
concur, about the robustness, but I think I will file this under 'if it ain't broke, don't fix it'